Source/WebKit/UIProcess/PDF/WKPDFPageNumberIndicator.mm

```diff
@@ -0,0 +1,207 @@
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "config.h"
+#import "WKPDFPageNumberIndicator.h"
+
+#if ENABLE(PDF_PAGE_NUMBER_INDICATOR)
+
+#import "PDFPluginIdentifier.h"
+#import "UIKitSPI.h"
+#import "WKWebViewIOS.h"
+#import <WebCore/GraphicsTypes.h>
+#import <WebCore/LocalizedStrings.h>
+#import <WebCore/PlatformCAFilters.h>
+#import <pal/spi/cocoa/QuartzCoreSPI.h>
+#import <wtf/Markable.h>
+#import <wtf/RetainPtr.h>
+#import <wtf/Seconds.h>
+#import <wtf/WeakObjCPtr.h>
+
+static constexpr CGFloat indicatorFontSize = 16;
+static constexpr CGFloat indicatorLabelOpacity = 0.4;
+static constexpr CGFloat indicatorCornerRadius = 7;
+static constexpr CGFloat indicatorMargin = 20;
+static constexpr CGFloat indicatorVerticalPadding = 6;
+static constexpr CGFloat indicatorHorizontalPadding = 10;
+
+static constexpr Seconds indicatorTimeout { 2_s };
+static constexpr Seconds indicatorFadeInDuration { 0.1_s };
+static constexpr Seconds indicatorFadeOutDuration { 0.75_s };
+static constexpr Seconds indicatorMoveDuration { 0.3_s };
+
+@implementation WKPDFPageNumberIndicator {
+    RetainPtr<UILabel> _label;
+    RetainPtr<UIVisualEffectView> _backdropView;
+    RetainPtr<NSTimer> _timer;
+    WeakObjCPtr<WKWebView> _webView;
+    Markable<WebKit::PDFPluginIdentifier> _pluginIdentifier;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame view:(WKWebView *)view pluginIdentifier:(WebKit::PDFPluginIdentifier)pluginIdentifier pageCount:(size_t)pageCount
+{
+    self = [super initWithFrame:frame];
+    if (!self)
+        return nil;
+
+    _pluginIdentifier = pluginIdentifier;
+    _webView = view;
+
+    [self setTranslatesAutoresizingMaskIntoConstraints:NO];
+    self.translatesAutoresizingMaskIntoConstraints = NO;
+
+    self.alpha = 0;
+    self.userInteractionEnabled = NO;
+    self.layer.allowsGroupOpacity = NO;
+    self.layer.allowsGroupBlending = NO;
+    [self setAutoresizingMask:UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight];
+
+    _backdropView = adoptNS([[UIVisualEffectView alloc] initWithEffect:[UIBlurEffect effectWithStyle:UIBlurEffectStyleLight]]);
+    [_backdropView setFrame:self.bounds];
+    [_backdropView setAutoresizingMask:UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight];
+    [[_backdropView layer] setCornerRadius:indicatorCornerRadius];
+    [_backdropView setClipsToBounds:YES];
+    [_backdropView setTranslatesAutoresizingMaskIntoConstraints:YES];
+    [self addSubview:_backdropView.get()];
+
+    _label = adoptNS([[UILabel alloc] initWithFrame:self.bounds]);
+    [_label setOpaque:NO];
+    [_label setBackgroundColor:nil];
+    [_label setTextAlignment:NSTextAlignmentCenter];
+    [_label setFont:[UIFont boldSystemFontOfSize:indicatorFontSize]];
+    [_label setTextColor:[UIColor blackColor]];
```

Comment:
Is it supposed to be black in both light and dark mode?

Reply:
(chatted about this in person, existing implementation is "always light mode", so this should be too)